src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/BuildUpToDateCheck.cs

```diff
@@ -308,30 +325,30 @@ private HashSet<string> CollectOutputs(TextWriter logger)
 
             var timestampCache = _fileTimestampCache.Value.TimestampCache ??
                     new Dictionary<string, DateTime>(StringComparer.OrdinalIgnoreCase);
-            var latestInput = GetLatestInput(logger, CollectInputs(logger), timestampCache);
-            var earliestOutput = GetEarliestOutput(logger, CollectOutputs(logger), timestampCache);
-
-            if (latestInput != null)
+            var latestInput = GetLatestInput(CollectInputs(logger), timestampCache);
+            var earliestOutput = GetEarliestOutput(CollectOutputs(logger), timestampCache);
+            
+            if (latestInput.time != null)
             {
-                Log(logger, "Lastest write timestamp on input is {0}.", latestInput.Value);
+                logger.Info("Lastest write timestamp on input is {0} on '{1}'.", latestInput.time.Value, latestInput.path);
             }
             else
             {
-                Log(logger, "One or more inputs do not exist.");
+                logger.Info("Input '{0}' does not exist.", latestInput.path);
             }
 
-            if (earliestOutput != null)
+            if (earliestOutput.time != null)
             {
-                Log(logger, "Earliest write timestamp on output is {0}.", earliestOutput.Value);
+                logger.Info("Earliest write timestamp on output is {0} on '{1}'.", earliestOutput.time.Value, earliestOutput.path);
             }
             else
             {
-                Log(logger, "One or more outputs do not exist.");
+                logger.Info("Output '{0}' does not exist.", earliestOutput.path);
             }
 
             // We are up to date if the earliest output write happened before the latest input write
```

Comment:
May be I am not reading this correct.

I would expect the condition to be, we are up to date if the latest input write is before the earliest output write? meaning, we are only update as long as all the outputs are created after last the input write has been made.